docs/02-guides/magic-button/contribution-mode.mdx

```diff
@@ -0,0 +1,132 @@
+---
+sidebar_position: 1
+title: Contribution Mode
+description:
+  Contribution Mode is the core of the Magic Button tool. When Contribution Mode
+  is enabled for a user, your storefront displays a set of tools created for
+  content creators.
+---
+
+import SinceVersion from "@site/src/components/SinceVersion";
+
+<SinceVersion tag="3.2" />
+
+<p>{frontMatter.description}</p>
+
+## API Reference
+
+The ContributionMode instance is exposed in the `user` context, see
+[app.user.contributionMode](/docs/remixed/api-reference/front-commerce-remix/front-commerce-app?tab=ContributionMode#appuser)
+for more information.
+
+## How to activate Contribution Mode
+
+The **Magic Button** is only visible when the **Contribution Mode** is active.
+
+The default activation for **Contribution Mode** is defined by the
+[`FRONT_COMMERCE_CONTRIBUTION_MODE_FORCE`](https://gitlab.blackswift.cloud/front-commerce/front-commerce/-/blob/bbaa7188cdd69e3fdca39859fc696d65019707b4/packages/core/contribution-mode/ContributionMode.ts#L23)
+environment variable
+
+:::warning
+
+This should only be used for development purposes.
+
+:::
+
+The `ContributionMode` state should be updated using the `setContributionMode`
+function.
+
+```ts
+user.contributionMode.setContributionMode({
+  enabled: true,
+  previewMode: true, // optional and will only change if contribution mode is enabled
+  xRayMode: true, // optional and will only change if contribution mode is enabled
+});
+```
+
+### Remix Action
```

Comment:
**thought:** these 3 sections show different ways to do the same thing. I think this will become more and more common as we build our public API on the main objects all around (`user`, `config`, `Services`…).

Since the examples are a bit redundant, I feel that readers will only really need one.

**suggestion:**  I suggest to replace these 3 headings with tabs. Each example will be collocated and readers could switch context easily depending on their use case at hand.

Reply:
Replaced with tabs.

![2023-12-13_10-50](https://github.com/front-commerce/developers.front-commerce.com/assets/75968/954a7944-44d7-4a4f-8d88-d3915760e1e1)